diff --git a/bi-dexhands/tasks/shadow_hand_pen.py b/bi-dexhands/tasks/shadow_hand_pen.py
index 4a9916e..3032d64 100644
--- a/bi-dexhands/tasks/shadow_hand_pen.py
+++ b/bi-dexhands/tasks/shadow_hand_pen.py
@@ -757,7 +757,6 @@ class ShadowHandPen(BaseTask):
         """
         num_ft_states = 13 * int(self.num_fingertips / 2)  # 65
         num_ft_force_torques = 6 * int(self.num_fingertips / 2)  # 30
-        print('dof: ', self.shadow_hand_dof_pos.shape, self.shadow_hand_another_dof_pos.shape)
         self.obs_buf[:, 0:self.num_shadow_hand_dofs] = unscale(self.shadow_hand_dof_pos,
                                                             self.shadow_hand_dof_lower_limits, self.shadow_hand_dof_upper_limits)
         self.obs_buf[:, self.num_shadow_hand_dofs:2*self.num_shadow_hand_dofs] = self.vel_obs_scale * self.shadow_hand_dof_vel
